cfme/common/provider.py

```diff
@@ -514,12 +442,22 @@ def get_credentials(cls, credential_dict, cred_type=None):
         """
         domain = credential_dict.get('domain', None)
         token = credential_dict.get('token', None)
-        return cls.Credential(
-            principal=credential_dict['username'],
-            secret=credential_dict['password'],
-            cred_type=cred_type,
-            domain=domain,
-            token=token)
+        if not cred_type:
```

Comment:
Jan, I'm going to move get_credentials to concrete provider class like I've made it for create and update methods of infrastructure providers. it will be done when I finish providers' conversion.